models/mock.go

```diff
@@ -0,0 +1,60 @@
+package models
+
+type MockDB struct {
+	Conversation *Conversation
+	Mapping      *UserConversationMapping
+	Mappings     []*UserConversationMapping
+	Error        error
```

Comment:
Is there a case where there's multiple DB methods being invoked in a handler and you would want to throw an error for one but not the others?

Reply:
Yeah, there probably will be. I don't think I actually even use this field at all yet. I'll just make it a list then, I guess, so it could be something like:
```
[nil, nil, nil, err]
```
if I want the 4th call to any DB method to return a specific error. Does that makes sense?

Reply:
Yeah I think that makes sense and there can be a counter in the struct to keep track of the calls.